src/quo2/components/list_items/saved_address/view.cljs

```diff
@@ -0,0 +1,69 @@
+(ns quo2.components.list-items.saved-address.view
+  (:require [quo2.components.avatars.wallet-user-avatar :as wallet-user-avatar]
+            [quo2.components.markdown.text :as text]
+            [quo2.foundations.colors :as colors]
+            [quo2.theme :as quo.theme]
+            [react-native.core :as rn]
+            [quo2.components.list-items.saved-address.style :as style]
+            [reagent.core :as reagent]
+            [quo2.components.icon :as icon]
+            [clojure.string :as string]))
+
+(defn- internal-view
+  []
+  (let [state       (reagent/atom :default)
+        active?     (atom false)
+        timer       (atom nil)
+        on-press-in (fn []
+                      (when-not (= @state :selected)
+                        (reset! timer (js/setTimeout #(reset! state :pressed) 100))))]
+    (fn [{:keys [blur? user-props customization-color
+                 on-press
+                 on-options-press
+                 theme]
+          :or   {customization-color :blue
+                 blur?               false}}]
+      (let [names        (string/split (:name user-props) " ")
```

Comment:
Note that `string/split` will return `[""]` when `user-props` is nil. I haven't checked if this is a problem, but it seems better if an empty vector was returned for your use case, otherwise a check like `(seq names)` would return true.

```clojure
(string/split nil " ") ; => [""]
```

Something like this may be better:

```clojure
(remove string/blank? (string/split nil " ")) ; => []
```

Reply:
Thanks for the suggestion, I added some extra validation because this suggestion made the avatar component to crash if nil is passed as first or last name